pipeline_steps/checkmarx.groovy

```diff
@@ -81,6 +82,10 @@ def scan(String scan_type, String repo_name, String exclude_folders){
                 throw e
             } //try
         } // retry
+
+        if (currentBuild.getRawBuild().getResult().isWorseThan(Result.fromString("SUCCESS"))) {
```

Comment:
If the scan is successful `currentBuild.getRawBuild().getResult() == null` which causes this line to raise an exception. I would suggest:
```suggestion
        if (Result.fromString(currentBuild.currentResult).isWorseThan(Result.fromString("SUCCESS"))) {
```